### PR TITLE
Add detail on map item grid

### LINKED ISSUE #4165: [needs-docs][layouts] Move grid "enabled" check box to grid properties
panel
Original commit: https://github.com/qgis/QGIS/commit/2d45eb5c70f3de4b343e3b2c11dea05e1815cb0d by nyalldawson

Allows it to sit next to the corresponding data defined button

### PR DESCRIPTION
**Pull request by DelazJ**
Fix #4165

### CODE REVIEW

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 1]
```diff
@@ -187,13 +187,16 @@ Double-click the added grid to rename it.
 
    Map Grids Dialog
 
-After you add a grid, you can activate the checkbox |checkbox| :guilabel:`Draw
-grid` to allow overlaying the grid onto the map item. Press the :guilabel:`Modify Grid...`
-button to access configuration options.
+After you add a grid, press the :guilabel:`Modify Grid...` button to access
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
To modify a grid, just select it and press the :guilabel:`Modify Grid...` button
to open the :guilabel:`Map Grid Properties` panel and access its
```

Comment (havatv) [reply source/docs/user_manual/print_composer/composer_items/composer_map.rst]: Different GUI in 3.4 and 3.10... (for the backporting)

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 2]
```diff
@@ -187,13 +187,16 @@ Double-click the added grid to rename it.
...
+configuration options.
 
 Grid Appearance
 ...............
 
+In the :guilabel:`Map Grid Properties` panel, check |checkbox|
+:guilabel:`Grid enabled` to allow the display of the grid onto the map item
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
 :guilabel:`Grid enabled` to show the grid on the map item.
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 3]
```diff
@@ -187,13 +187,16 @@ Double-click the added grid to rename it.
...
+and settings its properties.
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 4]
```diff
@@ -207,15 +210,18 @@ As grid type, you can specify to use a:
...
-* the :guilabel:`CRS` which could be different from the project CRS;
-* the :guilabel:`Interval units` to use for the grid references, in ``Map
-  Unit``, ``Fit Segment Width``, ``Millimeter`` or ``Centimeter``;
-* the :guilabel:`Interval` between two consecutive grid references in ``X``
-  and ``Y`` directions;
-* choosing ``Fit Segment Width`` will dynamically select the grid interval based 
-  on the map extent to a "pretty" interval. When selected, the ``Minimum`` and 
-  ``Maximum`` intervals can be set;
-* the :guilabel:`Offset` from the map item edges, in the ``X`` and/or the ``Y`` direction;
+* the :guilabel:`CRS` which could be different from the project CRS
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
* the :guilabel:`CRS`, which could be different from the project CRS
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 5]
```diff
@@ -207,15 +210,18 @@ As grid type, you can specify to use a:
...
+* the :guilabel:`Interval` type to use for the grid references. Available
+  options are ``Map Unit``, ``Fit Segment Width``, ``Millimeter`` or ``Centimeter``:
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]: "Fit Segment Width" is not an option in 3.4 (for backporting...)

Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
  options are:
  * ``Map Unit``
  * ``Fit Segment Width`` |310|
  * ``Millimeter``
  * ``Centimeter``
```

Comment (havatv) [reply source/docs/user_manual/print_composer/composer_items/composer_map.rst]: If you agree,

>.. |310| replace:: ``NEW in 3.10``

needs to be added to the substitution definitions at the bottom of the file...

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst [hunk 6]
```diff
@@ -207,15 +210,18 @@ As grid type, you can specify to use a:
...
+
+  * choosing ``Fit Segment Width`` will dynamically select the grid interval based 
+    on the map extent to a "pretty" interval. When selected, the ``Minimum`` and 
+    ``Maximum`` intervals can be set.
+  * the other options allow you to set the distance between two consecutive grid
+    references in ``X`` and ``Y`` directions.
```
Comment (havatv) [inline source/docs/user_manual/print_composer/composer_items/composer_map.rst]:
```suggestion
    references in the ``X`` and ``Y`` directions.
```

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst (hunk 5 above)
Comment (DelazJ) [reply source/docs/user_manual/print_composer/composer_items/composer_map.rst]: We should refrain adding it everywhere. Places where this is relevant imho are new expression functions, new algorithms or algs parameters and new server parameters; they have some impacts.
If we have to add it to any GUI option, features would become a nightmare to describe

**Comment by DelazJ** [comment]
@havatv I don't plan to backport to 3.4 because:
* #4165 is a 3.10 feature
* there is a previous PR that implemented the "fit segment width" property and a backport would require more tan a simple `git cherry-pick`
* the docs in 3.4 almost meet what the 3.4 GUI shows
* and last but not the least, 3.4 is no longer a target for me: it's at the end of its life and I don't want to spend/waste time on it and as already mentioned continue to break translators efforts.

File: source/docs/user_manual/print_composer/composer_items/composer_map.rst (hunk 5 above)
Comment (havatv) [reply source/docs/user_manual/print_composer/composer_items/composer_map.rst]: OK.

**Comment by havatv** [comment]
> @havatv I don't plan to backport to 3.4 because:
...

OK!

**Pull request merged**